services/src/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/FlowHistoryEvent.groovy

```diff
@@ -0,0 +1,16 @@
+package org.openkilda.functionaltests.helpers
+
+enum FlowHistoryEvent {
+
+    REROUTE_SUCCESS(FlowHistoryAction.FLOW_REROUTING, FlowHistoryAction.SET_THE_FLOW_STATUS_TO_UP),
+    REROUTE_FAIL(FlowHistoryAction.FLOW_REROUTING, FlowHistoryAction.SET_THE_FLOW_STATUS_TO_DOWN),
+    REROUTE_FAIL_PROTECTED_PATH(FlowHistoryAction.FLOW_REROUTING, FlowHistoryAction.FLOW_WAS_VALIDATED_SUCCESSFULLY),
```

Comment:
I am a bit confused how FLOW_REROUTING + FLOW_WAS_VALIDATED_SUCCESSFULLY is related to protected paths

Reply:
I just copied it from flowHistory output after flowState was changed to DEGRADED

Reply:
Why do you call it like that? It clearly means that `flow is validated after reroute` and I would agree with that, but the 'reroute fail for protected path' is just a special case of your test. The flow is validated, and you expect a degraded status in your exact special test, that's fine. But you cannot globally assume that 'rerouting + successfull validation' means 'reroute for protected path failed', that breaks laws of logic. Crocodiles are green, crocodiles are dangerous. Grass is green -> grass is dangerous?